packages/discord.js/src/managers/GuildEmojiManager.js

```diff
@@ -100,6 +100,44 @@ class GuildEmojiManager extends BaseGuildEmojiManager {
     for (const emoji of data) emojis.set(emoji.id, this._add(emoji, cache));
     return emojis;
   }
+
+  /**
+   * Deletes an emoji.
+   * @param {EmojiResolvable} emoji The Emoji resolvable to delete
+   * @param {string} [reason] Reason for deleting the emoji
+   * @returns {?Promise<void>}
+   */
+  async delete(emoji, reason) {
+    const id = this.resolveId(emoji);
+    if (!id) throw new TypeError('INVALID_TYPE', 'emoji', 'EmojiResolvable', true);
+    await this.client.api.guilds(this.guild.id).emojis(id).delete({ reason });
+  }
+
+  /**
+   * Edits an emoji.
+   * @param {EmojiResolvable} emoji The Emoji resolvable to edit
+   * @param {GuildEmojiEditData} data The new data for the emoji
+   * @param {string} [reason] Reason for editing this emoji
+   * @returns {Promise<GuildEmoji>}
+   */
+  async edit(emoji, data, reason) {
+    emoji = this.resolve(emoji);
+    if (!emoji) throw new TypeError('INVALID_TYPE', 'emoji', 'EmojiResolvable', true);
+    const roles = data.roles?.map(r => this.guild.roles.resolveId(r) ?? r);
```

Comment:
This should work
```suggestion
    const roles = data.roles?.map(this.guild.roles.resolveId);
```

Reply:
No, it won't due to unbound `this`.

Reply:
Well then at least the `?? r` can be removed, right?

Reply:
Yes, because `resolveId` already resolves an ID, so if it returns null... `r` is **definitely** not valid data

Reply:
Well then could you unresolve this?
```suggestion
    const roles = data.roles?.map(r => this.guild.roles.resolveId(r));
```